indico/modules/rb/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -458,20 +506,35 @@ class BookingDetails extends React.Component {
         )}
         {canReject && (
           <Popup trigger={rejectButton} position="bottom center" on="click">
-            <FinalForm onSubmit={data => this.changeState('reject', data)} render={renderForm} />
+            <FinalForm
+              onSubmit={data => this.changeState('reject', data)}
+              render={renderRejectionForm}
+            />
           </Popup>
         )}
         {showAccept && (
-          <Button
-            type="button"
-            icon="check circle"
-            color="green"
-            size="small"
-            onClick={() => this.changeState('approve')}
-            loading={actionInProgress === 'approve' && bookingStateChangeInProgress}
-            disabled={bookingStateChangeInProgress}
-            content={Translate.string('Accept booking')}
-          />
+          <Button.Group size="small" color="green">
+            <Button
+              type="button"
+              icon="check circle"
+              onClick={() => this.changeState('approve')}
+              loading={actionInProgress === 'approve' && bookingStateChangeInProgress}
+              disabled={bookingStateChangeInProgress || activeAcceptanceForm}
+              content={Translate.string('Accept booking')}
+            />
+            <Popup
+              trigger={acceptWithMessage}
```

Comment:
You could put the `<Button ... />` JSX directly in here, but nothing wrong with leaving it separate, especially since that's what we already do for `rejectionButton`.

I think the main reason why we have some of those buttons separate is actually that we weren't using prettier for code formatting back then, and it was really messy to have this kind of nesting at that point.